docs/profiler.md

```diff
@@ -1,136 +1,196 @@
+2. In the Powershell terminal, enter this command to run the profiler tool: `./profiler`.
 
-For accurate results set the path with `--data-file` to the same disk that will be used by the node, otherwise the results of the benchmark might not reflect the actual performance of the node. Please be warned that the contents of that file MIGHT be overwritten.
+Having run the tool, one may wonder, what does the output mean? Read on to understand the various input flags one can use to customize each profiler run and how to interpret the subsequent results.
 
-By default the profiler uses 1GiB sample data. This can be changed with the `--data-size` flag. Different sizes can be used to check for bottlenecks of the hardware.
+## Understanding the options and commands
 
-The duration of the benchmark can be set with `--duration` and defaults to 10 seconds. A longer duration will yield more accurate results.
+When running the profiler tool, you can provide several inputs to it. This section goes over the various commands and options for the profiler tool and what they mean.
 
-There are two parameters that can influence the proofing speed and can be optimized for go-spacemesh: `--threads` and `--nonces`.
-The general rule of thumb is the more threads the faster, but because different processors may have different behaviours when all threads are under load. Please check different values for `--threads` to find the optimal one.
-The more nonces are used with `--nonces` in ONE data pass, the bigger the chance to find a valid proof. The downside of using more `--nonces` is a heavier load on the CPU. It's important to note that nonces must be a multiple of 16.
+### Commands
 
-To simplify the estimation of nonces please use the following formula:
-in google sheets format
+* `proving`: We are telling the profiler to benchmark the entire PoST generation process.
+* `pow:`: We are telling the profiler to only benchmark the Proof of Work (PoW) part of the PoST generation. More on this below.
+
+If no command is given, the profiler runs the `proving` command by default.
+
+### Options
+
+* `--help`: Displays all the available commands and options along with their short descriptions. Use the `-h` flag for summarized help information.
+* `--data-file` (path): This is the path to an existing PoS file that we have already generated as a smesher during the [initialization or PoS generation](https://docs.spacemesh.io/docs/learn/post#generating-the-data) phase. By providing the path to an existing PoS data file, we are asking the profiler how long will it take to generate the PoST with the given PoS data. For accurate results, it is recommended to use the exact PoS data file on the same disk as used by the smeshing node. However, do note that the contents of the data file might get overwritten by the profiler tool.
```

Comment:
Advising the users to point to their real PoS data is dangerous as the profiler tool might truncate or extend the file (with random bytes) to match the `--data-size` value. Perhaps this behavior should change as it is quite a foot gun.

Users usually run profiler **before** initializing the PoS data to estimate the maximum data size they would be able to prove in the cycle gap window (12h on Spacemesh mainnet poets)

Reply:
Agree. I guess I can remove it from the text and later we can remove it from the code too?

Reply:
Perhaps it's best to remove it to avoid confusing users. I added `--data-file` to allow overwriting the default (`$TMP/file.bin`) in case the default didn't work for users. @nj Wdyt?

Reply:
We should keep it. The $TMP will likely be on a different drive/setup.

It's basically important that they use *same drive* for benchmark and real post data.

Reply:
I'd remove the part saying it is the path to the existing POS files. Users usually run the profiler **before initializing the data** and I'd emphasise what @pigmej said that it is important that `--data-file` is set to a filepath located on the same disk that the user plans to put the POS files on.